### PR TITLE
Separate VC runtime headers from CRT/PSDK headers

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

This PR separates VC runtime headers from CRT/PSDK headers. These headers come with VS, rather than with Windows SDK / CRT. Some of these are standard, like *intrin.h, others like sal.h or vcruntime.h are VS specific.
This separation allows for to import MS UCRT headers and in the future can allow to use the compiler provided headers, rather than shipping our own.

## Proposed changes

- Move compiler runtime headers into their own folder
- Add vcruntime.h and related (mostly empty)
- Move some definitions from crtdefs.h to vcruntime.h / vadefs.h
- Add some missing definitions
- Include vcruntime.h rather than crtdefs.h from a number of headers

## Tests
- ✅ KVM x86: https://reactos.org/testman/compare.php?ids=98567,98571,98574,98577
- ✅ KVM x64: https://reactos.org/testman/compare.php?ids=98569,98572,98575,98578

### CODE REVIEW

File: sdk/include/vcruntime/vcruntime.h [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+#endif // _HAS_NODISCARD
+
+#if defined _M_X64 || defined _M_ARM || defined _M_ARM64
+    #define _UNALIGNED __unaligned
+#else
+    #define _UNALIGNED
+#endif
+
+// FIXME: C++!
+#define __crt_countof(_Array) (sizeof(_Array) / sizeof(_Array[0]))
```
Comment (learn-more) [inline sdk/include/vcruntime/vcruntime.h:199]: Is this a FIXME for now, or for later.

**Review by binarymaster — APPROVED** [review]

**Review by HBelusca — APPROVED** [review]
Nerding time (for the 1st commit message):
> This separation will allow to use MS CRT headers.

--> "This separation will allow using MS CRT headers.", or: "This separation will allow MS CRT headers to be used."

> Eventually it can allow to compile with the compilers' runtime headers.

--> "Eventually it can allow compiling with ...", or: "... it can allow compilation with ..."

@cbialorucki @DarkFire01 or @Doug-Lyons may correct me if I'm wrong there :joy:

**Comment by HBelusca** [comment]
Question: where does the "core" CRT live now? in the vcruntime subdirectory?
(By "core CRT" I mean the very basic defines and the few functions of it.)

File: sdk/include/vcruntime/vcruntime.h (hunk 1 above)
Comment (tkreuzer) [reply sdk/include/vcruntime/vcruntime.h:199]: For later. I'll remove it.

Comment (tkreuzer) [reply sdk/include/vcruntime/vcruntime.h]: I added a proper C++ definiton.

File: sdk/include/vcruntime/vcruntime.h [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+ #ifndef _VCRT_DEFINED_CRTIMP
+  #define _VCRTIMP _CRTIMP
+ #elif defined(_VCRT_BUILD) && defined(CRTDLL) && !defined(_VCRT_SAT_1)
+  #define _VCRTIMP __declspec(dllexport)
+ #else
+  #define _VCRTIMP
+ #endif
+#endif
+
+#ifndef _CRT_STRINGIZE
```
Comment (learn-more) [inline sdk/include/vcruntime/vcruntime.h:51]: This is defined on line 8, and reason why it is a guard here?

File: sdk/include/vcruntime/vcruntime.h [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+# ifdef _CRT_SECURE_NO_DEPRECATE
+#  define _CRT_INSECURE_DEPRECATE(_Replacement)
+# else
+#  define _CRT_INSECURE_DEPRECATE(_Replacement) \
+    _CRT_DEPRECATE_TEXT("This may be unsafe, Try " #_Replacement " instead!")
+# endif
+#endif
+
+#ifndef _CRT_INSECURE_DEPRECATE_MEMORY
+#define _CRT_INSECURE_DEPRECATE_MEMORY(_Replacement)
```
Comment (learn-more) [inline sdk/include/vcruntime/vcruntime.h:98]: Any reason why this one is not implemented?
(I would guess too much spam?)

**Review by learn-more — APPROVED** [review]

**Review by DarkFire01 — APPROVED** [review]

File: sdk/include/vcruntime/vcruntime.h (hunk 2 above)
Comment (tkreuzer) [reply sdk/include/vcruntime/vcruntime.h]: That was an oversight. Fixed it.

**Comment by tkreuzer** [comment]
> Question: where does the "core" CRT live now? in the vcruntime subdirectory? (By "core CRT" I mean the very basic defines and the few functions of it.)

- vcruntime = core / compiler specific definitions and headers like sal.h and **intrin.h, comes with VS (stuff like **intrin.h is standardized and ships with other compilers as well), used by crt and some other headers
- crt (or ucrt): CRT implementation specific definitions, shipped separately (Windows SDK or mingw)

**Pull request merged**